src/reanimated2/fabricUtils.ts

```diff
@@ -1,33 +1,47 @@
 'use strict';
 /* eslint-disable camelcase */
 
-import { isFabric } from './PlatformChecker';
 import type { ShadowNodeWrapper } from './commonTypes';
 
-interface HostInstance {
-  _internalInstanceHandle: {
-    stateNode: {
-      node: ShadowNodeWrapper;
-    };
-  };
-}
-
-let findHostInstance_DEPRECATED: (ref: React.Component) => HostInstance;
+let findHostInstance_DEPRECATED: (ref: unknown) => void;
+let getInternalInstanceHandleFromPublicInstance: (ref: unknown) => {
+  stateNode: { node: unknown };
+};
 
 export function getShadowNodeWrapperFromRef(
   ref: React.Component
 ): ShadowNodeWrapper {
-  if (!findHostInstance_DEPRECATED && isFabric()) {
+  // load findHostInstance_DEPRECATED lazily because it may not be available before render
+  if (findHostInstance_DEPRECATED === undefined) {
     try {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
       findHostInstance_DEPRECATED =
-        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        // eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-unsafe-member-access
         require('react-native/Libraries/Renderer/shims/ReactFabric').findHostInstance_DEPRECATED;
     } catch (e) {
-      throw new Error(
-        '[Reanimated] Cannot import `findHostInstance_DEPRECATED`.'
-      );
+      findHostInstance_DEPRECATED = (_ref: unknown) => null;
```

Comment:
With this `findHostInstance_DEPRECATED` will not be reacquired. Is this the intent?

Reply:
I think so, if it is not present then it will not appear afterwards I think?